packages/plugins/accounts-manager/src/index.ts

```diff
@@ -82,4 +82,54 @@ export default class AccountsManager {
       this.logger.error(__("Error funding accounts"), err.message || err);
     }
   }
+
+  async getCoinbaseAddress(web3) {
+    const findAccountWithMostFunds = async () => {
+      const accounts = await web3.eth.getAccounts();
+      let highestBalance = {
+        balance: web3.utils.toBN(0),
+        account: ""
+      };
+      for (const account of accounts) {
+        // eslint-disable-next-line no-await-in-loop
+        let balance = await web3.eth.getBalance(account);
+        balance = web3.utils.toBN(balance);
+        if (balance.gt(highestBalance.balance)) {
+          highestBalance = { balance, account };
+        }
+      }
+      return highestBalance.account;
+    };
+
+    const findAlternativeCoinbase = async () => {
+      try {
+        return findAccountWithMostFunds();
+      } catch (err) {
+        throw new Error(`Error getting coinbase address: ${err.message || err}`);
+      }
+    };
+
+    try {
+      const coinbaseAddress = await web3.eth.getCoinbase();
+      // if the blockchain returns a zeroed address, we can find the account
+      // with the most funds and use that as the "from" account to txfer
+      // funds.
+      if (!coinbaseAddress ||
+        web3.utils.hexToNumberString(coinbaseAddress) === "0" || // matches 0x0 and 0x00000000000000000000000000000000000000
+        (await web3.eth.getBalance(coinbaseAddress)) === "0"
+      ) {
+        return await findAlternativeCoinbase();
+      }
+      return coinbaseAddress;
+    } catch (err) {
+      // if the blockchain doesn't support 'eth_coinbase' RPC commands,
+      // we can find the account with the most funds and use that as the
+      // "from" account to txfer funds.
+      if (err.message.includes("The method eth_coinbase does not exist/is not available")) {
+        const coinbase = await findAlternativeCoinbase();
+        return coinbase;
```

Comment:
Shall we just return `await findAlternativeCoinbase()` here?

Reply:
I modified to return the promise instead, ie `return this.findAlternativeCoinbase()`.